### ISSUE TITLE
iOS13 Image cannot be displayed, only the placeholder is displayed in iOS13

### ISSUE THREAD

**Issue by lanmeiceo** [SDWebImage/SDWebImage#2758]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 12.0 / 10.14.0 / 12.0 / 5.0
 SDWebImage Version      | e.g. 5.0.0 / 4.4.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 10 / Xcode 9
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by kinarob** [comment]
Such as: Image URL ? Platform Version ? SDWebImage Version? etc.. Thanks.

**Comment by lanmeiceo** [comment]
SDWebImage Verson is4.2.2
platform is iOS 13 ,Xcode Version  10
image url is http://testw.outdoorclub.com.cn/p3/club/activity/2019/6/14/9000331/345/15604787279917998.jpg
it can normally displayed in our intranet server,but iPhone cannot show.
![IMG_CF956FE4619D-1](https://user-images.githubusercontent.com/21143032/59558077-5ae5bb80-901c-11e9-81eb-2b03bdbd57fb.jpeg)

And,I find Douban app cannot shou picture normally in iOS 13,maybe is SDWebImage bug.
![IMG_1D8E3BD07DD8-1](https://user-images.githubusercontent.com/21143032/59558093-b152fa00-901c-11e9-9cd4-81a099b03d41.jpeg)

**Comment by kinarob** [comment]
@lanmeiceo  The Image URL
[http://testw.outdoorclub.com.cn/p3/club/activity/2019/6/14/9000331/345/15604787279917998.jpg](http://testw.outdoorclub.com.cn/p3/club/activity/2019/6/14/9000331/345/15604787279917998.jpg)
response is 404.

Could you provide a image URL response is 200 that can't be displayed in SDWebImage ?

**Comment by MrRightC** [comment]
I have updated to iOS13, only the gif image can't be displayed.

**Comment by dreampiggy** [comment]
@langyanduan If you say about WebP, it's alreadya issue and been fixed one year agon. See https://github.com/SDWebImage/SDWebImage/releases/tag/4.4.1 and upgrade to this version and above.

@MrRightC I test it locally and found no issue. Could you please provide a sample GIF url, and use our demo or your own demo to reproduce this issue ? If we can not reproduce, we can not provide a fix.

**Comment by MrRightC** [comment]
@dreampiggy Thanks, I've updated from 4.* -> 5.*, used the new API SDAnimatedImageView, now it works well.🤓👍

**Comment by koce** [comment]
@dreampiggy Hello,I have updated to iOS 13, use code " [imageView sd_setImageWithURL:placeholderImage:]" to load images. It works at first times, but it can't display GIF image when I refresh page. This may be Apple's bug?

**Comment by dreampiggy** [comment]
> @dreampiggy Hello,I have updated to iOS 13, use code " [imageView sd_setImageWithURL:placeholderImage:]" to load images. It works at first times, but it can't display GIF image when I refresh page. This may be Apple's bug?

What's your `imageView`. Is `UIImageView` or `SDAnimatedImageView` ?
And, please provide your demo GIF url, you can just run and test on our demo. Just clone the repo, run `pod install` and change the URL in code `MasterViewConrtroller.m`

**Comment by koce** [comment]
@dreampiggy Thank you, I have solved it by reuse UIImageView when I refresh that page rather than create new UIImageView.  Although I don't know why.

**Comment by dreampiggy** [comment]
@koce If you mixed use of `UIImageView` and `SDAnimatedImageView`. If one animated image using by `SDAnimatedImageView` is cached in memory (It's a `SDAnimatedImage` object), then another `UIImageView` can not display the animation, because the `SDAnimatedImage` can not play animation on normal imageView. Only the `UIAnimatedImage` can.

If you think you've faced this issue and want an official solution, we can decide to add a new option, like `SDWebImageMatchAnimatedImageClass`. Which will treat image class && image view class mismatch as a cache miss. So that your `UIImageView` will always create `UIAnimatedImage` but not `SDAnimatedImage` this object.

**Comment by koce** [comment]
@dreampiggy I only use UIImageView,  My project's SDWebImage version is very old, I find this problem when I updated to iOS 13, then I update SDWebImage  to the latest version, but it also not work, the UIImageView just display blank space.

**Comment by dreampiggy** [comment]
@koce Reproducable in our demo ?

Because I can not reproduce any above of issues. This is why I want to know how you use. What's your SDWebImage version.

**Comment by dreampiggy** [comment]
```
// In SDWebImage root directory
pod install
open SDWebImage.xcworkspace
// build and run
```

**Comment by koce** [comment]
Sorry, I try to make it appear in SDWebImage_iOS_Demo, but it always work well. I'll keep trying, I'll tell you when I succeed or find real reason.

**Comment by koce** [comment]
@dreampiggy 
Sorry, I try to make it appear in SDWebImage_iOS_Demo, but it always work well. I'll keep trying, I'll tell you when I succeed or find real reason.

**Comment by kinarobin** [comment]
@koce Are you make  SDWebImage_iOS_Demo in iOS13 device platform? You can try in iOS13 device platform.

**Comment by koce** [comment]
@kinarobin Yeah, I make it appear. [This Demo](https://github.com/koce/SDWebImageDemo), I try to modify DetailViewController.m, The GIF disappear when I click red button.

**Comment by koce** [comment]
@kinarobin Yeah, I make it appear. This Demo, I try to modify DetailViewController.m, The GIF disappear when I click red button.

System Version: iOS 13.0 beta 4
Xcode Version: 11.0 beta 4

**Comment by koce** [comment]
@kinarobin Yeah, I make it appear. [This Demo](https://github.com/koce/SDWebImageDemo), I try to modify DetailViewController.m, The GIF disappear when I click red button.

System Version: iOS 13.0 beta 4
Xcode Version: 11.0 beta

**Comment by kinarobin** [comment]
@koce From your demo, It 's you usage issue. 
 you should set frame before you call `sd_setImageWithURL :`
```ruby
CGFloat itemWidth = ([[UIScreen mainScreen] bounds].size.width - 10 * 5) / 4.0;
CGFloat itemHeight = itemWidth / 80.0 * 110.0;
self.imageView.frame = CGRectMake(0, 0, itemWidth, itemHeight);
// set frame at first
[self.imageView sd_setImageWithURL:[NSURL URLWithString:imageUrl] placeholderImage:nil];
```

**Comment by koce** [comment]
@kinarobin Okay, Thanks a lot for your help!👍

**Comment by dreampiggy** [comment]
Confirms to Apple's bug. See bugreport:

![image](https://user-images.githubusercontent.com/6919743/61625987-a3913400-acae-11e9-9c12-ead2100f1e7f.png)

Demo to trigger issue:
https://github.com/dreampiggy/UIAnimatedImageBugBehavior

**Comment by dreampiggy** [comment]
For current workaround, I can provide some suggestions:

1. Do not specify a zero frame before you load the image. Always set a frame or using Autolayout on your imageView.
2. Using the tricky code in my demo, set the UIImageView.image to nil and with previous image again.

**Comment by lanmeiceo** [comment]
I had sovled the problem

**Comment by BoySimonHu** [comment]
> I had sovled the problem

Can share your solution please？

**Comment by christiygag** [comment]
I fixed it by setting a frame that is greater than zero, like width = 0.5 and height = 0.5.

**Comment by dreampiggy** [comment]
> > I had sovled the problem
> 
> Can share your solution please？

Do not create `UIImageView` with zero frame. Create it with a (0,0,1,1) is OK.
